Library/Homebrew/utils/lock.sh

```diff
@@ -38,14 +38,19 @@ _create_lock() {
   local lock_fd="$1"
   local name="$2"
   local ruby="/usr/bin/ruby"
+  local python="/usr/bin/python"
   [[ -x "$ruby" ]] || ruby="$(which ruby 2>/dev/null)"
+  [[ -x "$python" ]] || python="$(which python 2>/dev/null)"
 
-  if [[ -n "$ruby" ]]
+  if [[ -x "$ruby" && $("$ruby" -e "puts RUBY_VERSION >= '1.8.7' ? 0 : 1") = 0 ]]
   then
     "$ruby" -e "File.new($lock_fd).flock(File::LOCK_EX | File::LOCK_NB) || exit(1)"
-  elif [[ -n "$(which flock)" ]]
+  elif [[ -x "$(which flock)" ]]
   then
     flock -n "$lock_fd"
+  elif [[ -x "$python" ]]
+  then
+    "$python" -c "import fcntl; fcntl.flock($lock_fd, fcntl.LOCK_EX | fcntl.LOCK_NB)"
```

Comment:
Yeah, it's available since at least 2.3 (which was shipped on Tiger), possibly older.